### PR TITLE
feat: Upload command

### LINKED ISSUE #962: The CLI can upload maps to the server
The old appland CLI tool (written in Golang) supports the `upload` command. We need to port that command (https://github.com/getappmap/appland-cli#upload) to the JS/TS CLI. We do not need to port the authentication or contexts features. Just expect APPLAND_API_KEY or APPMAP_API_KEY in the environment. stats has already been ported over.

### PR DESCRIPTION
**Pull request by dividedmind**
Adds upload command. Fixes #962 

Still no full feature-parity with the go cli, I'll keep adding to this or as new PRs if this has been merged.

### CODE REVIEW

File: packages/cli/src/cmds/upload.ts [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+  let total = paths.length;
+  const { baseURL } = loadConfiguration();
+  UI.progress(`Uploading ${total} AppMaps for ${app} to ${baseURL}...`);
+
+  const uuids: string[] = [];
+  let processed = 0;
+  for (const path of paths) {
+    processed += 1;
+    UI.status = `[${processed}/${total}] ${path}`;
+    const data = await readFile(path); //  TODO: stream this instead of slurping the whole thing?
```
Comment (kgilpin) [inline packages/cli/src/cmds/upload.ts:83]: One thing we do need to support in this v1 is the max file size parameter.

Comment (dividedmind) [reply packages/cli/src/cmds/upload.ts:83]: > One thing we do need to support in this v1 is the max file size parameter.

It isn't a parameter in the original :)

**Review by symwell — COMMENTED** [review]
In one of the runs I got this:
```
~/src/appmap-js/packages/cli$ npm start upload ~/src/forem/tmp/appmap

> @appland/appmap@3.64.2 start
> ts-node src/cli.ts upload /home/test/src/forem/tmp/appmap

Examining AppMaps...
ValidationError: "commit" mismatch ("68e79de6e400a5c5ec31287c68e49e287eb8a540" !== "19c6df892255e0ed11478705330c73e0ab4159be")
    at applyOrCheck (/home/test/src/appmap-js/packages/cli/src/cmds/upload.ts:26:13)
    at checkMetadata (/home/test/src/appmap-js/packages/cli/src/cmds/upload.ts:42:3)
    at collect (/home/test/src/appmap-js/packages/cli/src/cmds/upload.ts:54:5)
    at async Object.handler (/home/test/src/appmap-js/packages/cli/src/cmds/upload.ts:61:29)
```

Is this ok?  As a user I understand there's some mismatch but don't understand what I did wrong.  This is on `sw/use_appmap_to_create_openapi_doc` from https://github.com/symwell/forem

File: packages/cli/src/cmds/upload.ts (hunk 1 above)
Comment (kgilpin) [reply packages/cli/src/cmds/upload.ts:83]: Well, whatever you want to call this:

https://github.com/getappmap/appland-cli/blob/master/cmd/upload.go#L21

Comment (kgilpin) [reply packages/cli/src/cmds/upload.ts:83]: Otherwise the upload will be rejected by the server and it will fail without recourse.

**Review by kgilpin — CHANGES_REQUESTED** [review]
`-d` should be required to set the working directory. The `--appmap-dir` should specify the location of AppMaps relative to the working directory. If `--appmap-dir` is not specified, the default value is read from the _appmap.yml_. If no dir is specified, and no default is available, it's an error. This is the common pattern throughout the rest of the CLI, so for consistency, this command should work the same.

File: packages/cli/src/cmds/upload.ts [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+          MAX_APPMAP_SIZE / 1024
+        } KiB.`,
+        'Use --force if you want to upload it anyway.',
+      ].join('\n')
+    );
+}
+
+const command: CommandModule<{}, Arguments> = {
+  command: 'upload [appmap-dir]',
+  builder: {
```
Comment (kgilpin) [inline packages/cli/src/cmds/upload.ts:116]: Add option `-d` - working directory.

File: packages/cli/src/cmds/upload.ts [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+    appmapDir: {
```
Comment (kgilpin) [inline packages/cli/src/cmds/upload.ts:117]: Change to `--appmap-dir` option flag. Resolve relative to the working directory.

**Pull request merged**